### PR TITLE
Sample track activity indicator

### PR DESCRIPTION
**Pull request by Spekular**
Refactored version of #5307 by @serdnab (thanks!).

Reviewed(-ish) by myself during refactor, and quickly by @Veratil. I'll merge this in a few days if no on objects.

### CODE REVIEW

**Comment by Spekular** [comment]
Fully closes #1360 (listed in #1471).

File: src/tracks/SampleTrack.cpp [hunk 1]
```diff
@@ -835,6 +858,15 @@ SampleTrackView::SampleTrackView( SampleTrack * _t, TrackContainerView* tcv ) :
 
 
 
+void SampleTrackView::updateIndicator()
+{
+	if (model()->isPlaying()) m_activityIndicator->activateOnce();
```
Comment (Veratil) [inline src/tracks/SampleTrack.cpp:863]:
```suggestion
	if (model()->isPlaying()) { m_activityIndicator->activateOnce(); }
```

Not specifically needed, but always good to have.

File: src/tracks/SampleTrack.cpp [hunk 2]
```diff
@@ -826,6 +836,19 @@ SampleTrackView::SampleTrackView( SampleTrack * _t, TrackContainerView* tcv ) :
 	m_panningKnob->setLabel( tr( "PAN" ) );
 	m_panningKnob->show();
 
+	m_activityIndicator = new FadeButton(QApplication::palette().color(QPalette::Active,
+							QPalette::Background),
+						QApplication::palette().color(QPalette::Active,
+							QPalette::BrightText),
+						QApplication::palette().color(QPalette::Active,
+							QPalette::BrightText).darker(),
+						getTrackSettingsWidget());
```
Comment (Veratil) [inline src/tracks/SampleTrack.cpp:845]:
```suggestion
	m_activityIndicator = new FadeButton(
		QApplication::palette().color(QPalette::Active, QPalette::Background),
		QApplication::palette().color(QPalette::Active, QPalette::BrightText),
		QApplication::palette().color(QPalette::Active, QPalette::BrightText).darker(),
		getTrackSettingsWidget()
	);
```

I feel like this is easier to read.

File: src/tracks/SampleTrack.cpp [hunk 3]
```diff
@@ -592,7 +592,8 @@ SampleTrack::SampleTrack( TrackContainer* tc ) :
 	m_panningModel( DefaultPanning, PanningLeft, PanningRight, 0.1f,
 					this, tr( "Panning" ) ),
 	m_effectChannelModel( 0, 0, 0, this, tr( "FX channel" ) ),
-	m_audioPort( tr( "Sample track" ), true, &m_volumeModel, &m_panningModel, &m_mutedModel )
+	m_audioPort( tr( "Sample track" ), true, &m_volumeModel, &m_panningModel, &m_mutedModel ),
```
Comment (Veratil) [inline src/tracks/SampleTrack.cpp:595]:
```suggestion
	m_audioPort(tr("Sample track"), true, &m_volumeModel, &m_panningModel, &m_mutedModel),
```

File: src/core/Track.cpp [hunk 4]
```diff
@@ -3051,3 +3054,25 @@ void TrackView::createTCOView( TrackContentObject * tco )
...
+	FadeButton * actind = getActivityIndicator();
+	if (actind)
+	{
+		if (m_track->m_mutedModel.value())
+		{
+			actind->setActiveColor(QApplication::palette().color(
+							QPalette::Active,
+							QPalette::Highlight));
+		} else
+		{
```
Comment (Veratil) [inline src/core/Track.cpp:3072]:
```suggestion
		}
		else
		{
```

File: src/core/Track.cpp [hunk 5]
```diff
@@ -3051,3 +3054,25 @@ void TrackView::createTCOView( TrackContentObject * tco )
...
+void TrackView::muteChanged()
+{
+	FadeButton * actind = getActivityIndicator();
+	if (actind)
+	{
+		if (m_track->m_mutedModel.value())
+		{
+			actind->setActiveColor(QApplication::palette().color(
+							QPalette::Active,
+							QPalette::Highlight));
```
Comment (Veratil) [inline src/core/Track.cpp:3070]:
```suggestion
			actind->setActiveColor(
				QApplication::palette().color(QPalette::Active, QPalette::Highlight)
			);
```

Easier to read?

File: src/core/Track.cpp [hunk 6]
```diff
@@ -3051,3 +3054,25 @@ void TrackView::createTCOView( TrackContentObject * tco )
...
+		} else
+		{
+			actind->setActiveColor(QApplication::palette().color(
+							QPalette::Active,
+							QPalette::BrightText));
```
Comment (Veratil) [inline src/core/Track.cpp:3075]:
```suggestion
			actind->setActiveColor(
				QApplication::palette().color(QPalette::Active, QPalette::BrightText)
			);
```

Easier to read?

**Comment by Veratil** [comment]
Tested and works as advertised. 👍

File: src/core/Track.cpp (hunk 5 above)
Comment (Spekular) [reply src/core/Track.cpp:3070]: Some other alternatives, any preference?

1. Same initial indentation level, put color() arguments on the same line

```suggestion
			actind->setActiveColor(QApplication::palette().color(
				QPalette::Active, QPalette::Highlight
			));
```

2. Start the function with an `if (!actind) { return; }` statement so the calls can be indented by one level less, put the entire argument to setActiveColor on it's own line

```suggestion
		actind->setActiveColor(
			QApplication::palette().color(QPalette::Active, QPalette::Highlight)
		);
```

Comment (Veratil) [reply src/core/Track.cpp]: I don't feel like 1 is as readable. 2 is essentially the same as my suggestion. :)

File: src/tracks/SampleTrack.cpp [hunk 7]
```diff
@@ -826,6 +834,18 @@ SampleTrackView::SampleTrackView( SampleTrack * _t, TrackContainerView* tcv ) :
 	m_panningKnob->setLabel( tr( "PAN" ) );
 	m_panningKnob->show();
 
+	m_activityIndicator = new FadeButton(
+		QApplication::palette().color(QPalette::Active, QPalette::Background),
+		QApplication::palette().color(QPalette::Active, QPalette::BrightText),
+		QApplication::palette().color(QPalette::Active, QPalette::BrightText).darker(),
+		getTrackSettingsWidget()
+	);
+	m_activityIndicator->setGeometry(settingsWidgetWidth-2*24-11, 2, 8, 28);
```
Comment (Veratil) [inline src/tracks/SampleTrack.cpp:843]:
```suggestion
	m_activityIndicator->setGeometry(settingsWidgetWidth - 2 * 24 - 11, 2, 8, 28);
```

**Comment by Spekular** [comment]
Forgot to hit squash-and-merge, sorry!

**Pull request merged**